examples/multipart-form/main.rs

```diff
@@ -0,0 +1,73 @@
+#![feature(async_await, futures_api)]
+
+#[macro_use]
+extern crate serde_derive;
+
+use http::status::StatusCode;
+use std::io::Read;
+use tide::{body, App};
+
+#[derive(Serialize, Deserialize, Clone)]
+struct Message {
+    key1: Option<String>,
+    key2: Option<String>,
+    file: Option<String>,
+}
+
+async fn upload_file(
+    multipart_form: body::MultipartForm,
+) -> Result<body::Json<Message>, StatusCode> {
+    // https://stackoverflow.com/questions/43424982/how-to-parse-multipart-forms-using-abonander-multipart-with-rocket
+    let mut message = Message {
+        key1: None,
+        key2: None,
+        file: None,
+    };
+
+    let mut multipart = multipart_form.0;
+    multipart
+        .foreach_entry(|mut entry| match entry.headers.name.as_str() {
+            "file" => {
+                let mut vec = Vec::new();
+                entry.data.read_to_end(&mut vec).expect("can't read");
+                message.file = String::from_utf8(vec).ok();
+                println!("key file got");
+            }
+
+            "key1" => {
+                let mut vec = Vec::new();
+                entry.data.read_to_end(&mut vec).expect("can't read");
+                message.key1 = String::from_utf8(vec).ok();
+                println!("key1 got");
+            }
+
+            "key2" => {
+                let mut vec = Vec::new();
+                entry.data.read_to_end(&mut vec).expect("can't read");
+                message.key2 = String::from_utf8(vec).ok();
+                println!("key2 got");
+            }
+
+            _ => {
+                println!("key neglected");
+            }
+        })
+        .expect("Unable to iterate multipart?"); // TODO: why an unspecified key will break the foreach_entry() call??? See test curl example below.
```

Comment:
`foreach_entry` will always try to read multipart boundary. If we don't read the data, the cursor will remain same, so it fails to read boundary. Seems like this is a bug of `multipart` crate.

Reply:
🤔 Perhaps I can go ask around on there repo?

Reply:
It turns out to be a bug. https://github.com/abonander/multipart/issues/114

Reading entry.data again in our example to make the example work now.